bsdbpf/bsd_bpf_sniffer.go

```diff
@@ -171,6 +172,9 @@ func (b *BPFSniffer) pickBpfDevice() {
 			break
 		}
 	}
+	if b.options.BPFDeviceName == "" {
```

Comment:
I don't believe this check does what you want it to.  Note that after the first iteration of the loop, BPFDeviceName is no longer empty, due to line 169.  So this if statement will never run.

Instead, how about changing the 'break' at line 172 to a 'return', then just panicing if you get past the for loop?

Or, better yet, make this function return an error, and check it at line 100, returning the error if it's non-nil.